src/native/_auth.js

```diff
@@ -214,7 +215,13 @@ module.exports = {
     access_container_fetch: helpers.Promisified(null, [ref.refType(ContainerPermissions), t.usize], (args) => {
       const ptr = args[0];
       const len = args[1];
-      let arrPtr = ref.reinterpret(ptr, ContainerPermissions.size * len);
+      let arrPtr;
+      try {
+        arrPtr = ref.reinterpret(ptr, ContainerPermissions.size * len);
+      } catch (err) {
+        const noContainersErr = errConst.NO_CONTAINERS;
+        return makeFfiError(noContainersErr.code, noContainersErr.msg); 
```

Comment:
I don't think we should be generating an error in this case, what I get is that the safe_app lib returns a `null` pointer if the app doesn't have permissions for any container, so we should be making sure to handle that scenario and and generating an empty list of containers.
So I think here instead of doing the try/catch you just need to check for `len > 0` before trying to read the array, and that should be enough, the rest is covered by the `for` loop below.

Reply:
Right, this makes much more sense and makes things simpler. Thank you.